### DISCUSSION TITLE
Are prompt templates required for ggml models?

### DISCUSSION THREAD

**Discussion by RahulVivekNair** [ggml-org/llama.cpp#1980] [Q&A]
I currently run every model with -i -ins which seems to be using the alpaca format. I see on the model pages though that most models have different templates such as ' USER: ' 'ASSISTANT:' etc. Do I need to and how do I use these in llama.cpp?

Also vicuna 1.1 onwards they mention a different prompt separator '</s>', do I somehow mention this too for llama.cpp?

**Comment by mirek190** [comment]
I think you can use --in-prefix and --in-suffix for it

**Reply by RahulVivekNair** [reply]
Thank you ,this works!Do you know if it's required for these ggml models though,I feel like the answers and chat are better but it might be placebo. Also do you know anything about the separators like ```</s>```used in vicuna?
